effect/default_effector.go

```diff
@@ -26,55 +24,21 @@ func NewDefaultEffector() *DefaultEffector {
 	return &e
 }
 
-// MergeEffects merges all matching results collected by the enforcer into a single decision.
-func (e *DefaultEffector) MergeEffects(expr string, effects []Effect, results []float64) (bool, int, error) {
-	result := false
-	explainIndex := -1
-	if expr == "some(where (p_eft == allow))" {
-		result = false
-		for i, eft := range effects {
-			if eft == Allow {
-				result = true
-				explainIndex = i
-				break
-			}
-		}
-	} else if expr == "!some(where (p_eft == deny))" {
-		result = true
-		for i, eft := range effects {
-			if eft == Deny {
-				result = false
-				explainIndex = i
-				break
-			}
-		}
-	} else if expr == "some(where (p_eft == allow)) && !some(where (p_eft == deny))" {
-		result = false
-		for i, eft := range effects {
-			if eft == Allow {
-				result = true
-			} else if eft == Deny {
-				result = false
-				explainIndex = i
-				break
-			}
-		}
-	} else if expr == "priority(p_eft) || deny" {
-		result = false
-		for i, eft := range effects {
-			if eft != Indeterminate {
-				if eft == Allow {
-					result = true
-				} else {
-					result = false
-				}
-				explainIndex = i
-				break
-			}
-		}
+func (e *DefaultEffector) NewStream(expr string, cap int) (DefaultEffectorStream) {
```

Comment:
It must returns `EffectorStream` interface.  @hsluoyz Should we avoid using `painc` ?